libs/gltfio/include/gltfio/MaterialProvider.h

```diff
@@ -36,8 +36,13 @@ enum class AlphaMode : uint8_t {
 
 // The following struct gets hashed so all padding bits should be explicit.
 // Tell the compiler to emit a warning if it adds any padding.
+#if defined(__clang__)
 #pragma clang diagnostic push
 #pragma clang diagnostic warning "-Wpadded"
+#elif defined(_MSC_VER)
+#pragma warning( push )
+#pragma warning(disable : 4324)
```

Comment:
Since there are two instances of this, maybe define a UTILS_DISABLE_WARNING_PADDED (or a better name)?

Reply:
I agree, if it's doable, it would probably be cleaner and less error prone in the future.

Reply:
Sounds good, see `fb7c465c5d06411b2845e02397c98f8fa8493d74`. The only drawback is that if you compile with /W0 in MSVC, `#pragma warning(1: XXXX)` will do nothing 😞 . This is different in Clang, where even -Wno-everything is overridden by `clang diagnostic warning "-Wpadded"`. You can consider raising an error instead of a warning because both MSVC and Clang ignore compiler flags when a pragma requests an error for a warning.

Note: at first I misunderstood the clang code and disabled the warning on MSVC, instead of enabling it. Fixed that too